python_modules/dagster/dagster/components/resolved/model.py

```diff
@@ -83,6 +85,33 @@ def from_model(fn: Callable[["ResolutionContext", Any], Any], **kwargs):
         """Resolve this field by invoking the function which will receive the entire parent model."""
         return Resolver(ParentFn(fn), **kwargs)
 
+    @staticmethod
+    def union(*resolvers: "Resolver"):
+        def _resolve_fn(context: "ResolutionContext", field_value: Any):
+            accumulated_errors = []
+            for r in resolvers:
+                try:
+                    result = r.fn.callable(context, field_value)
+                    if result is not None:
+                        return result
+                except Exception:
+                    accumulated_errors.append(traceback.format_exc())
+
+            raise ResolutionException(
+                "No resolver matched the field value"
+                + "\n"
+                + textwrap.indent(
+                    "\n".join(accumulated_errors),
+                    prefix="  ",
+                ),
+            )
```

Comment:
maybe `functools.partial` a free standing function instead if it just needs `resolvers`. Feel its nice to avoid inline fn defs when its easy